### PR TITLE
Create 0152-Maximum-Product-Subarray.md

### LINKED ISSUE #699: Add leetcode solution for problem 152
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

Add leetcode solution for problem maximum product subarray

### Suggested Change

.

### Rationale

_No response_

### Urgency

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by mahek0620**
## Related Issue

Closes #699 

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [X] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: dsa-solutions/lc-solutions/0100-0199/0152-Maximum-Product-Subarray.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Input: nums = [-2,0,-1]
+Output: 0
+Explanation: The result cannot be 2, because [-2,-1] is not a subarray.
+```
+
+### Constraints
+
+- $1 <= nums.length <= 2 * 10**4$
+- $10 <= nums[i] <= 10$
+- $The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0152-Maximum-Product-Subarray.md:43]: replace `$The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.$` to `The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.`

File: dsa-solutions/lc-solutions/0100-0199/0152-Maximum-Product-Subarray.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        return result;
+    }
+};
+
+
+    ```
+
+  </TabItem>
+</Tabs>
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0152-Maximum-Product-Subarray.md:149]: remove

```md

</TabItem>

</Tabs>
```

because you used 2 times

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
now update your code based on suggestions

**Comment by mahek0620** [comment]
@Ajay-Dhangar The code has been modified

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Comment by Ajay-Dhangar** [comment]
@mahek0620, to calculate the score for your leaderboard, you'll need to add something to those files and then try it out at the bottom. Below is the code you need to include for your author profile in the documentation:

```md
---

<h2>Authors:</h2>

<div style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', gap: '10px'}}>
{['github_username'].map(username => (
 <Author key={username} username={username} />
))}
</div>
```

**Pull request merged**